pkg/yang/entry.go

```diff
@@ -675,34 +727,120 @@ func ToEntry(n Node) (e *Entry) {
 				e.merge(nil, nil, ToEntry(a))
 			}
 		case "type":
-			// We don't expect this to happen, so throw an error.
-			// BUG(borman): I think a deviate statement might trigger this.
-			e.addError(fmt.Errorf("%s: unexpected type in %s:%s", Source(n), n.Kind(), n.NName()))
+			// The type keyword is specific to deviate to change a type. Other type handling
+			// (e.g., leaf type resolution) is done outside of this case.
+			n, ok := n.(*Deviate)
+			if !ok {
+				e.addError(fmt.Errorf("unexpected type found, only valid under Deviate, is %T", n))
+				continue
+			}
 
+			if n.Type != nil {
+				if errs := n.Type.resolve(); errs != nil {
```

Comment:
overly cautious question: could we refer to a type in an earlier `deviate` statement, which we then, in a later deviation, set via another deviate stmt to `not-supported`?